packages/bpk-docs/src/index.js

```diff
@@ -17,67 +17,93 @@
  */
 
 import React from 'react';
+import PropTypes from 'prop-types';
+import 'bpk-stylesheets/base';
 import ReactDOM from 'react-dom';
+import 'bpk-stylesheets/base.css';
 import Helmet from 'react-helmet';
 import ReactDOMServer from 'react-dom/server';
 import {
-  Router,
-  RouterContext,
-  match,
-  browserHistory,
-  createMemoryHistory,
-} from 'react-router';
-import 'es6-promise/auto';
-
-import routes from './routes';
+  BrowserRouter,
+  StaticRouter,
+  withRouter,
+  matchPath,
+} from 'react-router-dom';
+
+import Routes, { ROUTES_MAPPINGS } from './routes';
 import template from './template';
 import { extractAssets } from './webpackStats';
 
+const ScrollToTop = withRouter(
+  class extends React.Component {
+    static propTypes = {
+      children: PropTypes.node.isRequired,
+      location: PropTypes.object.isRequired, // eslint-disable-line
+    };
+
+    componentDidUpdate(prevProps) {
+      if (this.props.location !== prevProps.location) {
+        window.scrollTo(0, 0);
+      }
+    }
+
+    render() {
+      return this.props.children;
+    }
+  },
+);
+
 if (typeof window !== 'undefined' && typeof document !== 'undefined') {
   const root = document.getElementById('react-mount');
 
   ReactDOM.render(
-    React.createElement(Router, {
-      history: browserHistory,
-      onUpdate: () => {
-        if (`${window.location}`.indexOf('#') === -1) {
-          window.scrollTo(0, 0);
-        }
-      },
-      routes,
-    }),
+    <BrowserRouter>
+      <ScrollToTop>
+        <Routes />
+      </ScrollToTop>
+    </BrowserRouter>,
     root,
   );
 }
 
-export default (locals, callback) => {
-  const history = createMemoryHistory();
-  const location = history.createLocation(locals.path);
-  const assets = extractAssets(locals.webpackStats);
+export default (() => {
+  const flattenRoutes = routes =>
+    routes.reduce((all, route) => {
+      all.push(route);
 
-  match({ routes, location, history }, (error, redirectLocation, props) => {
-    // Explicit check for null here due to odd behaviour with react router's match function
-    // It passes undefined in cases where matches are not found.
-    // So we use their error object if it is truthy, otherwise we create our own.
-    if (error !== null) {
-      return callback(
-        error ||
-          new Error(`React Router failed to match ${JSON.stringify(location)}`),
-      );
+      if (route.routes) {
+        return all.concat(flattenRoutes(route.routes));
```

Comment:
maybe spread here rather than concat, it'll make us like the cool kids

Reply:
Hmm, I actually prefer `concat` here than `[...all, ...flattenRoutes(route.routes)]` because I'll have to `spread` twice. I generally go with spread operator only when I'm creating a new literal array and want to merge another one in, e.g. `[1, 2, 3, ...extra]`

Reply:
but it's sooo ES5 😄

Reply:
And it turns out concat is much faster in this case :). https://jsperf.com/concat-vs-spread1
<img width="967" alt="screen shot 2018-08-23 at 10 52 01" src="https://user-images.githubusercontent.com/1457263/44518619-a4969b80-a6c2-11e8-9090-45d282f21cdc.png">